src/config_parse.c

```diff
@@ -475,6 +475,10 @@ int git_config_parse(
 		size_t line_len = parser->ctx.line_len;
 		char c;
 
+		/*
+		 * Get either first non-whitespace character or, if that does
+		 * not exist, the first whitespace character
```

Comment:
`because we need to preserve whitespace when writing back` ?

Maybe `We skip over whitespace until we find something meaningful, otherwise there's only whitespace and we need to preserve it when writing back`.

Reply:
True, that would certainly go a long way in explaining what happens here. Thanks, will amend!